commands/initcmd.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package commands
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/GoogleContainerTools/kpt/pkg/kptfile"
+	"github.com/GoogleContainerTools/kpt/pkg/kptfile/kptfileutil"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"sigs.k8s.io/cli-utils/pkg/common"
+	"sigs.k8s.io/cli-utils/pkg/config"
+)
+
+const defaultInventoryName = "inventory"
+
+// KptInitOptions encapsulates fields for kpt init command. This init command
+// fills in inventory values in the Kptfile.
+type KptInitOptions struct {
+	dir                string // Directory with Kptfile
+	force              bool   // Set inventory values even if already set in Kptfile
+	name               string // Inventory object name
+	namespace          string // Inventory object namespace
+	inventoryID        string // Inventory object unique identifier label
+	defaultInitOptions *config.InitOptions
+}
+
+// NewKptInitOptions returns a pointer to an initial KptInitOptions structure.
+func NewKptInitOptions(f cmdutil.Factory, ioStreams genericclioptions.IOStreams) *KptInitOptions {
+	io := config.NewInitOptions(f, ioStreams)
```

Comment:
The naming here is a little confusing. Can we chose a different name on this variable than we use in the function definitions. It is the same name, but different types.

Reply:
Fixed.